vcstools/prof_utils.py

```diff
@@ -555,6 +578,470 @@ def check_clip(prof_to_check, toomuch=0.8, toolittle_frac=0., toolittle_absolute
+                     linewidth=3, linestyle="-"))
+        labels = ["Maxima", "Noise estimate bounds",
+                  "Profile", "Smoothed Profile", "Error"]
+        plt.legend(lines, labels)
+        plt.xlabel("Bins")
+        plt.ylabel("Amplitude")
+        plt.title(plot_name[:-4])
+        plt.savefig(plot_name, bbox_inches="tight")
+        plt.close()
+
+    comp_est_dict = {}
+    comp_est_dict["maxima"] = real_max
+    comp_est_dict["underest_on_pulse"] = underest_on_pulse
+    comp_est_dict["overest_on_pulse"] = overest_on_pulse
+    comp_est_dict["underest_off_pulse"] = underest_off_pulse
+    comp_est_dict["overest_off_pulse"] = overest_off_pulse
+    comp_est_dict["norm_noise_std"] = off_pulse_noise_std
+    comp_est_dict["norm_noise_mean"] = off_pulse_noise_mean
+    comp_est_dict["alpha"] = alpha
+
+    return comp_est_dict
+
+
+def get_off_pulse(on_pulse_pairs):
+    """ 
+    Works out the off pulse ranges from the on-pulse
+
+    Parameters:
+    -----------
+    on_pulse_pairs: list
+        List of sub-lists/tuples. Each sub-list is a pair of numbers that describes the upper and lower bounds of the 
+        ON PULSE region
+
+    Returns:
+    --------
+    off_pulse: list
+        A list of two-lists that decribes the off-pulse region
+    """
+    off_pulse = []
+    # Make cycling generators to deal with wrap-around
+    current_pair_gen = itertools.cycle(on_pulse_pairs)
+    next_pair_gen = itertools.cycle(on_pulse_pairs)
+    next(next_pair_gen)
+    # Figure out the off-pulse profile
+    for i in range(len(on_pulse_pairs)):
+        current_pair = next(current_pair_gen)
+        next_pair = next(next_pair_gen)
+        off_pulse.append([current_pair[1], next_pair[0]])
+    return off_pulse
+
+
+def profile_region_from_pairs(profile, pairs):
+    """
+    Acquires a list of the region described by a pair of indexes accounting for wrap-around
+
+    Parameters:
+    -----------
+    profile: list
+        The profile that we want to get the region of
+    pairs: list/tuple
+        A list of lists/tuples of length two that describes the indexes of the region(s) to grab
+
+    Return: 
+    -------
+    region: list
+        The region of the profile described by the pairs
+    """
+    profile = list(profile)
+    region = []
+    for pair in pairs:
+        if pair[0] < pair[1]:
+            region = profile[pair[0]:pair[1]]
+        else:
+            region = profile[pair[0]:]
+            region.extend(profile[:pair[1]])
+    return region
+
+
+def filled_profile_region_between_pairs(profile, pairs, fill_value=0):
+    """
+    Fills the off-pair region of a profile with the fill value
+
+    Parameters:
+    -----------
+    profile: list/numpy array
+        The profile we want to fill
+    pairs: list/tuple
+        A list of lists/tupls of length two that describes the region we DON'T want to fill
+    fill_value: float
+        The value to fill the region with. Default=0.
```

Comment:
If I'm understanding this description then a better name might be filled_profile_region_outside_pairs

Reply:
I disagree. You are filling the region between the sets of pairs that you're handing the function.

Reply:
Then maybe filled_profile_between_outside_pair_sets